sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/spanner/SpannerIO.java

```diff
@@ -190,6 +190,19 @@ public static CreateTransaction createTransaction() {
   }
 
   /**
+   * Returns a {@link DoFn} that can be applied to a {@link PCollection<ReadOperation>}.
+   * @param config a spanner configuration.
+   * @param transaction a PCollectionView with a transaction, usually created with
+   * {@link Read#createTransaction()}. If null, the doesn't provide a transactional guarantees.
+   * @return a {@link DoFn} that concurrently reads from multiple Cloud Spanner sources.
+   */
+  @Experimental
+  public static DoFn<ReadOperation, Struct> readFn(SpannerConfig config,
```

Comment:
DoFn's should not be part of the public API of a PTransform https://beam.apache.org/contribute/ptransform-style-guide/#exposing-a-ptransform-vs-something-else

Instead, please follow the example of TextIO.readAll() https://github.com/apache/beam/blob/master/sdks/java/core/src/main/java/org/apache/beam/sdk/io/TextIO.java#L170 and create a new PTransform SpannerIO.readAll()